core/services/ocr2/delegate.go

```diff
@@ -986,6 +1002,21 @@ func (d *Delegate) newServicesOCR2Functions(
 		ReportingPluginFactory:       nil, // To be set by NewFunctionsServices
 	}
 
+	s4OracleArgs := libocr2.OCR2OracleArgs{
```

Comment:
Do you think we could wrap the creation of this struct in a function too, like you did with the Provider above? We could also consider moving it inside NewFunctionsServices() ...

Reply:
We won't save much of lines I think.. but loose in readability.